librad/src/net/peer.rs

```diff
@@ -503,13 +520,29 @@ where
 {
     type Update = Gossip;
 
-    async fn put(&self, provider: PeerId, has: Self::Update) -> PutResult {
+    async fn put(&self, provider: PeerId, mut has: Self::Update) -> PutResult<Self::Update> {
         let span = tracing::info_span!("Peer::LocalStorage::put");
         let _guard = span.enter();
 
+        let git = self.inner.get().await.unwrap();
+        if git.peer_id() == provider {
```

Comment:
Ah wait, no, this is not enough anyway -- the `has` could contain our own `PeerId`. I think the previous version was probably better, as it will do the right thing even if this is our own `has` somehow relayed back to us.

Reply:
Ok :sweat_smile: